src/analysis_and_optimization/Debug_data_generation.mli

```diff
@@ -1,5 +1,13 @@
+open Core_kernel
 open Middle
 
+val json_to_mir :
+     (Expr.Typed.t SizedType.t * 'a * string) list
+  -> Yojson.Basic.t
+  -> (string, Expr.Typed.t) Map.Poly.t
+
 val gen_values_json :
-     (Expr.Typed.t SizedType.t * Expr.Typed.t Transformation.t * string) list
-  -> string
+     ?filter:bool
```

Comment:
Mind adding a brief doc comment for both these functions? Particularly for things like what `filter` does.

Otherwise I'm pretty happy with this PR as it stands. It's a nice fix and improved functionality.